### PR TITLE
#18 add inbox forwarding api

### LINKED ISSUE #18: Add inbox api

### PR DESCRIPTION
**Pull request by asmaNode**

### CODE REVIEW

**Review by rezk2ll — DISMISSED** [review]

File: src/index.js [hunk 1]
```diff
@@ -3,11 +3,13 @@ import applicationDiscoveryServiceApi from './api/core/application-discovery-ser
...
+import inboxAPi from './api/inbox';
 
 export {
   Client,
   applicationDiscoveryServiceApi,
   contactApi,
   davImportApi,
-  jamesApi
+  jamesApi,
+  inboxAPi
```
Comment (chamerling) [inline src/index.js]: `inboxApi`

**Review by rezk2ll — DISMISSED** [review]

**Review by rezk2ll — DISMISSED** [review]

File: src/api/inbox/inbox-forwarding-user.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const BASE_PATH = '/unifiedinbox/api/inbox';
+
+export default function(client) {
+  return {
+    updateForwardingConfigurations
+  };
+
+  /**
+   * Update forwarding configurations
```
Comment (tuanlc) [inline src/api/inbox/inbox-forwarding-user.js:9]: JSdoc for a function should describe params & return

File: src/api/inbox/inbox-forwarding-user.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+   */
+  function updateForwardingConfigurations(domainId, configurations) {
+    return client.api({
+      url: `${BASE_PATH}/forwardings/configurations?scope=domain&domain_id=${domainId}`,
+      method: 'put',
+      data: { ...configurations }
```
Comment (tuanlc) [inline src/api/inbox/inbox-forwarding-user.js]: Why not simpler like this:

```suggestion
      data: configurations
```

**Review by rezk2ll — DISMISSED** [review]

File: src/api/inbox/inbox-forwarding-user.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+export default function(client) {
+  return {
+    updateForwardingConfigurations
+  };
+/**
+ * Update forwarding configurations
+ * @param {String} domainId target domain ID
+ * @param {Object} configurations target configurations
+ */
+  function updateForwardingConfigurations(domainId, configurations) {
```
Comment (itswillta) [inline src/api/inbox/inbox-forwarding-user.js]:
```suggestion

  /**
   * Update forwarding configurations
   * @param {String} domainId target domain ID
   * @param {Object} configurations target configurations
   */
  function updateForwardingConfigurations(domainId, configurations) {
```

**Review by rezk2ll — APPROVED** [review]

File: src/api/inbox/inbox-forwarding-user.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    updateForwardingConfigurations
+  };
+
+  /**
+   * Update forwarding configurations
+   *
+   * @param {String} domainId target domain ID
+   * @param {Object} configurations target configurations
+   */
+
```
Comment (itswillta) [inline src/api/inbox/inbox-forwarding-user.js:14]:
```suggestion
```

:trollface:

File: test/api/inbox/inbox-forwarding-user.spec.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+describe('The inbox APIs', () => {
+  let inboxForwarding;
+
+  beforeEach(() => {
+    inboxForwarding = inboxApi(setup.client).inboxForwarding;
+  });
+
+  describe('The updateForwardingConfigurations function', () => {
+    it('should send PUT request to the right endpoint to update inbox configurations', function(done) {
```
Comment (itswillta) [inline test/api/inbox/inbox-forwarding-user.spec.js:12]:
```suggestion
    it('should send PUT request to the right endpoint to update inbox configurations', done => {
```

Comment (itswillta) [reply test/api/inbox/inbox-forwarding-user.spec.js:12]: It's suggested that we use arrow functions in Jest tests.

**Review by tuanlc — APPROVED** [review]

**Pull request merged**